### Q.1) A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


### ANSWER:

In [1]:
import pandas as pd
import numpy as np
import scipy

In [3]:
#STEP1: Define hypothesis
#X1= mean of unit A
#X2= mean of unit B

#Null Hypothesis = H0 =X1-X2=o     ##There is a significant difference in the diameter of the cutlet between two points.
#Alternative Hypothesis = H1= X1-X2=! 0   ##There is no  significant difference in the diameter of the cutlet between two points.


In [2]:
#STEP2: Identify Z-test or t-test

In [4]:
#Given Data

In [11]:
Cutlets=pd.read_csv("Cutlets.csv")

In [13]:
Cutlets.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [14]:
# Here two sample are given , hence two sample t-test is used 

In [16]:
#Sample1
A =Cutlets.iloc[:,0]

In [17]:
#Sample2
B =Cutlets.iloc[:,1]

In [19]:
Cutlets.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [30]:
#Parameters 
X1=7.01909
X2=6.964297
S1=0.288408
S2=0.343401

#Degrees of Freedom = n1+n2-2
n1=35
n2=35
dof=35+35-2
'\u03B1' #Unicode for alpha
α=0.05


In [32]:
import sys
import unicodedata

In [33]:
t=(X1-X2)/ np.sqrt((S1*S1/n1)+S2*S2/n2)

In [35]:
t

0.7228519750859105

In [36]:
X1-X2

0.05479300000000009

In [37]:
np.sqrt((S1**2+S2**2)/n1)

0.0758011347945587

In [39]:
from scipy import stats
tcri = stats.t.ppf(0.05,68)

In [40]:
tcri

-1.6675722806611732

In [ ]:
#Here t > tcri, Hence we conclude that we fail to reject null hypothesis

In [41]:
#2nd Method
#For Two Sample Test
from scipy import stats

In [42]:
T=stats.ttest_ind(A,B)

In [43]:
T

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.4722394724599501)

In [13]:
if pvalue <0.05: #alpha value is 0.05 or 5%
    print("We are rejecting Null Hypothesis")
else:
    print("WE are accepting Null Hypothesis")

WE are accepting Null Hypothesis


## Conclusion 

###  Here p > α (0.4722>0.05) Hence we fail to reject null hpothesis.
### There is no  significant difference in the diameter of the cutlet between two points.


###   Q2) A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

## ANSWER:

In [2]:
#STEP1: Define Hypothesis
#μ1=Mean of laboratory 1
#μ2=Mean of laboratory 2
#μ3=Mean of laboratory 3
#μ4=Mean of laboratory 4

#NULL HYPOTHESIS: H0 : μ1=μ2=μ3=μ4.  (Groups means(μ) are equal )
#ALTERNATIVE HYPOTHESIS: H1: All μ are not equal.

In [1]:
# unicode for greek values
print('\u03BC')

μ


In [6]:
#Load data file.
import pandas as pd
df=pd.read_csv("labTAT.csv")

In [7]:
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [11]:
import scipy.stats as stats
Fvalue, pvalue = stats.f_oneway(df['Laboratory 1'],df['Laboratory 2'],df['Laboratory 3'],df['Laboratory 4'])
# Seeing if the overall model is significant
print('F-Statistic=%.5f, p=%.5f' % (F, p))

F-Statistic=118.70422, p=0.00000


In [12]:
if pvalue <0.05: #alpha value is 0.05 or 5%
    print("We are rejecting Null Hypothesis")
else:
    print("WE are accepting Null Hypothesis")

We are rejecting Null Hypothesis


### CONCLUSION:

Here p value is less than 0.05. Hence we are rejecting null hypothesis.

### This implies that TAT for at least one laboratory is different.

### Q3)       Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

### ANSWER:

In [14]:
#Define Hypothesis
#H0: There is no diff in between male and female buyer ratio across all the regions
#Ha: There is  diff in between male and female buyer ratio at least in one region

#μe= mean of male and female buyer ratio for east regions
#μw= mean of male and female buyer ratio for west regions
#μn= mean of male and female buyer ratio for north regions
#μs= mean of male and female buyer ratio for south regions
# Null hypothesis H0 : μe=μs=μn=μw
#Alternative hypothesisHa: all means are not equal

In [ ]:
#Loading Data

In [20]:
dfq3 = pd.read_csv("BuyerRatio.csv")

In [22]:
dfq3

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [23]:
import scipy.stats as stats
# stats f_oneway functions takes the groups as input and returns ANOVA F and p value
fvalue, pvalue = stats.f_oneway(dfq3['East'],dfq3['West'],dfq3['North'],dfq3['South'])
print("F value is",fvalue,"and","P value is" ,pvalue)

F value is 0.3068565327605033 and P value is 0.8204060154887464


In [12]:
if pvalue <0.05: #alpha value is 0.05 or 5%
    print("We are rejecting Null Hypothesis")
else:
    print("WE are accepting Null Hypothesis")

WE are accepting Null Hypothesis


## Conclusion

### #Here p value is greater than 0.05, hence we fail to reject null hypothesis

# This implies that there is no difference between male and female buyer ratio across all the regions

### Q4)      TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


## ANSWER:

### The test is applied when you have two categorical variables from a single population. It is used to determine whether there is a significant association between the two variables.

In [1]:
#Step 1: Define Hypothesis
#H1: The average number of error free customer order from are equal across all the center
#H2: The average number of error free customer order from are different across all the center

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import scipy.stats as stats


In [3]:
dfq4= pd.read_csv("Costomer+OrderForm.csv")

In [4]:
dfq4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [5]:
data = pd.get_dummies(dfq4, columns=dfq4.columns)

In [6]:
data.head()

Phillippines_Defective  Phillippines_Error Free  Indonesia_Defective  \
0                       0                        1                    0   
1                       0                        1                    0   
2                       0                        1                    1   
3                       0                        1                    0   
4                       0                        1                    0   

   Indonesia_Error Free  Malta_Defective  Malta_Error Free  India_Defective  \
0                     1                1                 0                0   
1                     1                0                 1                1   
2                     0                1                 0                0   
3                     1                0                 1                0   
4                     1                1                 0                0   

   India_Error Free  
0                 1  
1                 0  
2                 1  
3                 1  
4                 1

In [7]:
A1=dfq4["Phillippines"].replace({"Error Free": "1", "Defective": "0"})
A2=dfq4["Indonesia"].replace({"Error Free": "1", "Defective": "0"})
A3=dfq4["Malta"].replace({"Error Free": "1", "Defective": "0"})
A4=dfq4["India"].replace({"Error Free": "1", "Defective": "0"})
                                 

In [8]:
A2

0      1
1      1
2      0
3      1
4      1
      ..
295    1
296    1
297    1
298    1
299    0
Name: Indonesia, Length: 300, dtype: object

In [10]:
import scipy.stats as stats
# stats f_oneway functions takes the groups as input and returns ANOVA F and p value
fvalue, pvalue = stats.f_oneway(A1,A2,A3,A4)
print("F value is",fvalue,"and","P value is" ,pvalue)

F value is 1.286168556089167 and P value is 0.2776780955705948


In [11]:
if pvalue <0.05: #alpha value is 0.05 or 5%
    print("We are rejecting Null Hypothesis")
else:
    print("WE are accepting Null Hypothesis")

WE are accepting Null Hypothesis


# Conclusion

### #Here p value is greater than 0.05. Hence we reject null hypothesis

## This implies that the average number of error free customer order forms are equal across all the centers